tests/end-to-end/harbor-api.bats

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bats
+
+# Test using Harbor via the API
+
+setup_file() {
+  load "../common/lib"
+  load "../common/lib/harbor"
+
+  harbor_load_env
+
+  export harbor_endpoint
+
+  export project="end-to-end-test-project"
+  export robot="end-to-end-test-robot"
+
+  export robot_id_path="/tmp/compliantkubernetes-apps-end-to-end-test-robot-id"
+  export robot_name_path="/tmp/compliantkubernetes-apps-end-to-end-test-robot-name"
+  export robot_secret_path="/tmp/compliantkubernetes-apps-end-to-end-test-robot-secret"
```

Comment:
Worth using `tempfile` to pick these names? Trades risk of multiple things using the same files for a risk of a failed test leaving them behind if the tests is somehow interrupted before they're cleaned up.

Reply:
That is why I'm using the slug thing, I should perhaps move that to the env load and require it there to make it safer.

The reason why I wanted it to be the same across runs is that then if the process gets killed for some reason it may still successfully clean it up the next run.

Perhaps I should just rework the clean up function do a recursive deletion instead. (The setup will still fail early as it will have conflict if anything is already setup.)

Reply:
Unsure how this interacts with bats own cleanup but using [bash traps](https://stackoverflow.com/a/41619743) is one way:
```bash
cleanup () {
rm "/tmp/some-file"
}
trap cleanup EXIT
```